eppo-sdk-test/validators/BanditEvaluatorTest.cs

```diff
@@ -0,0 +1,396 @@
+        });
+    }
+
+
+    [Test]
+    public void ShouldEquallyWeighEvenField()
+    {
+        var scores = new List<ActionScore> {
+           new("action1", 0.5),
+           new("action2", 0.5),
+           new("action3", 0.5),
+           new("action4", 0.5),
+        };
+        var expectedWeights = new List<ActionScore> {
+           new("action1", 1.0/4),
+           new("action2", 1.0/4),
+           new("action3", 1.0/4),
+           new("action4", 1.0/4)
+        };
+        var gamma = 0.1;
+        var minProbability = 0.1;
+
+        var weights = banditEvaluator.WeighActions(scores, gamma, minProbability);
+        AssertActionScoreListsMatch(weights, expectedWeights);
+    }
+
+    [Test]
+    public void TestEvaluateBandit()
+    {
+        // Mock data
+
+        const string flagKey = "test_flag";
+        const string subjectKey = "test_subject";
+        var subjectAttributes = new ContextAttributes(subjectKey)
+        {
+             { "age", 25.0 } ,
+             { "location", "US" }
+        };
+
+        ContextAttributes[] actionContexts = {
+            new("action1") { { "price", 10.0 },{ "category", "A" } },
+            new("action2") { { "price", 20.0 },{ "category", "B" } }
+        };
+
+        var coefficients = new Dictionary<string, ActionCoefficients>()
+        {
+            {
+                "action1", new ActionCoefficients("action1", 0.5)
+                {
+                    SubjectNumericCoefficients = new List<NumericAttributeCoefficient>() { new("age", 0.1, 0.0) },
+                    SubjectCategoricalCoefficients = new List<CategoricalAttributeCoefficient>() 
+                    {
+                        new( "location",  0.0, new DoubleDictionary() { { "US", 0.2 } } )
+                    },
+                    ActionNumericCoefficients = new List<NumericAttributeCoefficient>() { new( "price", 0.05,  0.0 )},
+                    ActionCategoricalCoefficients = new List<CategoricalAttributeCoefficient>() 
+                    {
+                        new( "category",  0.0, new DoubleDictionary(){ { "A", 0.3 } } )
+                    }
+                }
+            },
+            {
+                "action2", new ActionCoefficients("action2", 0.3)
+                {
+                    SubjectNumericCoefficients = new List<NumericAttributeCoefficient>() { new("age", 0.1, 0.0) },
+                    SubjectCategoricalCoefficients = new List<CategoricalAttributeCoefficient>()
+                    {
+                        new( "location",  0.0, new DoubleDictionary() { { "US", 0.2 } } )
+                    },
+                    ActionNumericCoefficients = new List<NumericAttributeCoefficient>() { new( "price", 0.05,  0.0 )},
+                    ActionCategoricalCoefficients = new List<CategoricalAttributeCoefficient>() 
+                    {
+                        new( "category",  0.0, new DoubleDictionary(){ { "A", 0.3 } } )
```

Comment:
This seems to be `B` in the [Python test](https://github.com/Eppo-exp/python-sdk/blob/main/test/bandit_test.py#L295).

Reply:
[DeterministicSharder strikes again](https://github.com/Eppo-exp/python-sdk/blob/102bd6fdc03db4d3b41cce2a5214b03d6183cc43/test/bandit_test.py#L307)

Reply:
good catch, thanks!